src/plugins/implicitRelationships/index.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2023 Vendicated and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByProps, findStoreLazy } from "@webpack";
+import { ChannelStore, FluxDispatcher, GuildStore, RelationshipStore, SnowflakeUtils, UserStore } from "@webpack/common";
+import { User } from "discord-types/general";
+import { Settings } from "Vencord";
+
+const UserAffinitiesStore = findStoreLazy("UserAffinitiesStore");
+
+interface UserAffinity {
+    user_id: string;
+    affinity: number;
+}
+
+export default definePlugin({
+    name: "ImplicitRelationships",
+    description: "Shows your implicit relationships in the Friends tab.",
+    authors: [Devs.Dolfies],
+    patches: [
+        // Counts header
+        {
+            find: ".FRIENDS_ALL_HEADER",
+            replacement: {
+                match: /toString\(\)\}\);case (\i\.\i)\.BLOCKED/,
+                replace: 'toString()});case $1.IMPLICIT:return "Implicit — "+arguments[1];case $1.BLOCKED'
+            },
+        },
+        // No friends page
+        {
+            find: "FriendsEmptyState: Invalid empty state",
+            replacement: {
+                match: /case (\i\.\i)\.ONLINE:return (\i)\.SECTION_ONLINE/,
```

Comment:
if you don't actually need that part for anything, you can just slap it in a lookaround so you dont match it
```suggestion
                match: /case (\i\.\i)\.ONLINE:(?=return (\i)\.SECTION_ONLINE)/,
```